nabs/callbacks.py

```diff
@@ -165,3 +166,60 @@ def stop(self, doc):
                 tags=['run_table', 'RE'],
                 title='run_table'
             )
+
+
+class BECOptionsPerRun(BestEffortCallback):
+    """
+    Callback that reads options from metadata to decide whether or not
+    to plot/print table.
+
+    Settings should only persist for one run, after which the settings
+    are reverted.  Options can be set by passing metadata to the RE:
+
+    .. code-block: python
+
+        # Can pass as metadata in the plan
+        RE(bp.scan([], motor, -1, 1, 5, md={'disable_plots': True}))
+
+        # Or as kwargs to the RunEngine
+        RE(plan(), disable_table=False)
+
+    """
+    # Metadata key : bec attribute
+    valid_keys = {'disable_plots': '_plots_enabled',
+                  'disable_table': '_table_enabled'}
+
+    def __init__(self, *args, **kwargs):
+        self.prev_settings = {}
+        super().__init__(*args, **kwargs)
+
+    def start(self, doc):
```

Comment:
This is really clean and should absolutely work as advertised in all normal cases.

Is it all possible to have a glitchy case where:
- We start a run with disable plots
- Something crashes and end_run is never called despite our best efforts in our finalize wrapper constructs
- The plots are still disabled for all scans that follow

Reply:
And if that is possible, what's the best way for us to deal with it? I'm not immediately sure.

Reply:
I believe in that case, the user could simply invoke `bec.enable_plots()` and the next time a start doc is seen, that will plots enabled will be the most recent "previous setting"

Reply:
Oh sure I see.  I think we can easily restore settings if they still exist in the start document.  Modification incoming

Reply:
Oh I like that solution- simple and clean